arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/LowGasStipendPrecompileCallTests.java

```diff
@@ -0,0 +1,327 @@
+      for (ArgumentCase argumentCase : ArgumentCase.values()) {
+        arguments.add(Arguments.of(ECREC, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(SHA256, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(RIPEMD160, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(ID, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(ALTBN128_ADD, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(ALTBN128_MUL, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(Address.ALTBN128_PAIRING, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(BLAKE2B_F_COMPRESSION, argumentCase, gasCase, false));
+      }
+      // The NON_ZERO for MODEXP case will be treated in a separate test
+      arguments.add(Arguments.of(MODEXP, ArgumentCase.ZERO, gasCase, false));
+      arguments.add(Arguments.of(MODEXP, ArgumentCase.ZERO, gasCase, true));
+    }
+    return arguments.stream();
+  }
+
+  // Support methods
+  private static void prepareBlake2F(
+      BytecodeCompiler program, int rLeadingByte, int callDataOffset) {
+    program
+        .push(rLeadingByte) // For simplicity, we only set the first byte of r
+        .push(callDataOffset + 2) // offset
+        // Writing rLeadingByte at this offset
+        // allows to have r = 0x00000000 or r = 0x00001200
+        .op(OpCode.MSTORE8);
+  }
+
+  private void prepareSha256Ripemd160Id(BytecodeCompiler program, int nWords, int callDataOffset) {
+    populateMemory(program, nWords, callDataOffset);
+  }
+
+  private static BigInteger prepareModexp(
+      int bbs, int mbs, int ebs, int callDataSize, BytecodeCompiler program) {
+    final Bytes32 bbsPadded = Bytes32.leftPad(Bytes.of(bbs));
+    final Bytes32 ebsPadded = Bytes32.leftPad(Bytes.of(ebs));
+    final Bytes32 mbsPadded = Bytes32.leftPad(Bytes.of(mbs));
+    final Bytes32 bemPadded =
+        Bytes32.rightPad(
+            Bytes.fromHexString("0x" + "aa".repeat(bbs) + "ff".repeat(ebs) + "bb".repeat(mbs)));
+    // Note that is an arbitrary value respecting bbs, ebs, mbs
+
+    program
+        .push(bbsPadded)
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(ebsPadded)
+        .push(32) // offset
+        .op(OpCode.MSTORE)
+        .push(mbsPadded)
+        .push(64) // offset
+        .op(OpCode.MSTORE)
+        .push(bemPadded)
+        .push(96) // offset
+        .op(OpCode.MSTORE);
+
+    // This is computed here for convenience, and it is used for pricing the MODEXP precompile
+    return OobOperation.computeExponentLog(
+            Bytes.concatenate(bbsPadded, ebsPadded, mbsPadded, bemPadded),
+            BigInteger.valueOf(callDataSize),
+            BigInteger.valueOf(bbs),
+            BigInteger.valueOf(ebs),
+            BigInteger.valueOf(mbs))
+        .max(BigInteger.ONE);
+  }
+
+  /**
+   * Computes the returnAtCapacity based on the precompile address, and callDataSize in the case of
```

Comment:
`ReturnAtCapacity` isn't something one computes, it's something one is free to set as one pleases in a CALL-type instruction. It seems to me you are computing the `ReturnDataSize` of (successful) calls to the various precompiles. These aren't up for debate, they are indeed determined by the precompile.